yt_dlp/downloader/hls.py

```diff
@@ -8,153 +8,64 @@
 from .fragment import FragmentFD
 from .. import webvtt
 from ..dependencies import Cryptodome_AES
-from ..utils import bug_reports_message, parse_m3u8_attributes, update_url_query
+from ..utils import ExtractorError, bug_reports_message, parse_m3u8_attributes, update_url_query
 
 
-class HlsFD(FragmentFD):
-    """
-    Download segments in a m3u8 manifest. External downloaders can take over
-    the fragment downloads by supporting the 'm3u8_frag_urls' protocol and
-    re-defining 'supports_manifest' function
-    """
-
-    FD_NAME = 'hlsnative'
+class InitializationFragmentError(ExtractorError):
+    def __init__(self):
+        super().__init__(
+            'Initialization fragment found after media fragments, unable to download', expected=True)
 
-    @staticmethod
-    def can_download(manifest, info_dict, allow_unplayable_formats=False):
-        UNSUPPORTED_FEATURES = [
-            # r'#EXT-X-BYTERANGE',  # playlists composed of byte ranges of media files [2]
 
-            # Live streams heuristic does not always work (e.g. geo restricted to Germany
-            # http://hls-geo.daserste.de/i/videoportal/Film/c_620000/622873/format,716451,716457,716450,716458,716459,.mp4.csmil/index_4_av.m3u8?null=0)
-            # r'#EXT-X-MEDIA-SEQUENCE:(?!0$)',  # live streams [3]
+class HlsMediaManifest:
```

Comment:
Everything's a static method? So you are basically using it as a namespace? perhaps this should become a submodule of utils.

Alternatively, we could go the OOP-ish route and intialize an instance with the manifest 
contents. E.g. `HLSMediaManifest(s, url).get_fragments(...)`

Either way, this should go in utils imo. Eventually, we'd want `InfoExtractor._parse_m3u8_formats_and_subtitles` and friends to also be moved into the same namespace